### PR TITLE
Add Makefile

### PR DESCRIPTION
**Pull request by KhaledHegazy222**
Added `makefile` to streamline common tasks including:
- build
- run
- run unit tests
- run integration tests
- vet

completes `makefile` task in #13

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+GO_CMD = go
+BINARY_NAME = catalyst
```
Comment (AbdelrahmanElawady) [inline Makefile:2]: let's write the binary to bin/ directory and add it in .gitignore

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+build:
+	$(GO_CMD) build -o $(BINARY_NAME) main.go
+
+run:
+	$(GO_CMD) run main.go
+
+vet:
+	$(GO_CMD) vet ./...
+
+unit_test:
```
Comment (AbdelrahmanElawady) [inline Makefile:15]: also you might want to check out Makefile target naming convention [here](https://stackoverflow.com/questions/32130664/is-there-a-naming-convention-for-makefile-targets-and-variables).

File: .gitignore [hunk 3]
```diff
@@ -1 +1,2 @@
 **/.DS_Store
+bin/
```
Comment (AbdelrahmanElawady) [inline .gitignore:2]: last thing, files should end in new line

**Review by AbdelrahmanElawady — APPROVED** [review]
Looks good

**Review by tareknaser — APPROVED** [review]
Thanks

**Pull request merged**